lib/devise_saml_authenticatable/model.rb

```diff
@@ -52,7 +55,7 @@ def authenticate_with_saml(saml_response, relay_state)
           end
 
           if Devise.saml_update_user || (resource.new_record? && Devise.saml_create_user)
-            set_user_saml_attributes(resource, attributes)
+            Devise.saml_update_resource_hook.call(resource, decorated_response)
             if (Devise.saml_use_subject)
               resource.send "#{key}=", auth_value
             end
```

Comment:
Seems like this (and the following `save!` call) should go within the block as well.